### PR TITLE
Scaffold

### PR DESCRIPTION
**Pull request by RomanSipyak**

### CODE REVIEW

File: src/Reenbit.ChuckNorris.DataAccess/ReenbitChuckNorrisDbContext.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Reenbit.ChuckNorris.DataAccess
+{
+    class ReenbitChuckNorrisDbContext : DbContext
+    {
+        public ReenbitChuckNorrisDbContext(DbContextOptions<ReenbitChuckNorrisDbContext> options) : base(options) 
+        {
+
```
Comment (tabas) [inline src/Reenbit.ChuckNorris.DataAccess/ReenbitChuckNorrisDbContext.cs:13]: delete empty line

File: src/Reenbit.ChuckNorris.DataBase/JokeCategory.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿CREATE TABLE [dbo].[JokeCategory]
+( 
+    [Joke_Id] INT NOT NULL,
```
Comment (tabas) [inline src/Reenbit.ChuckNorris.DataBase/JokeCategory.sql:3]: rename JokeId

File: src/Reenbit.ChuckNorris.DataBase/JokeCategory.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿CREATE TABLE [dbo].[JokeCategory]
+( 
+    [Joke_Id] INT NOT NULL, 
+    [Category_Id] INT NOT NULL,
```
Comment (tabas) [inline src/Reenbit.ChuckNorris.DataBase/JokeCategory.sql:4]: CategoryId

File: src/Reenbit.ChuckNorris.DataBase/Jokes.sql [hunk 4]
```diff
@@ -0,0 +1,8 @@
+﻿CREATE TABLE [dbo].[Jokes]
+(
+	[Id] INT IDENTITY(1,1) NOT NULL, 
+    [IconUrl] NVARCHAR(255) NULL,
```
Comment (tabas) [inline src/Reenbit.ChuckNorris.DataBase/Jokes.sql:4]: 255->2048

File: src/Reenbit.ChuckNorris.DataBase/Jokes.sql [hunk 5]
```diff
@@ -0,0 +1,8 @@
+﻿CREATE TABLE [dbo].[Jokes]
+(
+	[Id] INT IDENTITY(1,1) NOT NULL, 
+    [IconUrl] NVARCHAR(255) NULL, 
+    [Url] NVARCHAR(255) NULL,
```
Comment (tabas) [inline src/Reenbit.ChuckNorris.DataBase/Jokes.sql:5]: 255->2048

File: src/Reenbit.ChuckNorris.Domain/Entities/JokeCategory.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+using System.Text;
+
+namespace Reenbit.ChuckNorris.Domain.Entities
+{
+    public class JokeCategory
+    {
+        public int Id { get; set; }
+
+        public int JokeId { get; set; }
+        public Joke Joke { get; set; }
```
Comment (tabas) [inline src/Reenbit.ChuckNorris.Domain/Entities/JokeCategory.cs:12]: each property should be separated by new line

File: src/Reenbit.ChuckNorris.DataAccess/Mapping/JokeMap.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using Reenbit.ChuckNorris.Domain.Entities;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Reenbit.ChuckNorris.DataAccess.Mapping
+{
+    public class JokeMap : IEntityTypeConfiguration<Joke>
```
Comment (tabas) [inline src/Reenbit.ChuckNorris.DataAccess/Mapping/JokeMap.cs:10]: internal, this class will be used only inside the current library

File: src/Reenbit.ChuckNorris.API/Extentions/DependencyInjectionConfig.cs [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+{
+    public static class DependencyInjectionConfig
+    {
+        public static IServiceCollection RegisterDependencies(this IServiceCollection services)
+        {
+            RegisterInfrastructure(services);
+            RegisterDataAccess(services);
+            return services;
+        }
+
```
Comment (tabas) [inline src/Reenbit.ChuckNorris.API/Extentions/DependencyInjectionConfig.cs:20]: -

File: src/Reenbit.ChuckNorris.API/Extentions/DependencyInjectionConfig.cs [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        public static void RegisterDataAccess(this IServiceCollection services)
+        {
+            services.AddScoped<IUnitOfWorkFactory, UnitOfWorkFactory>();
+        }
+
+        public static void RegisterInfrastructure(this IServiceCollection services)
+        {
+            services.AddTransient<IConfigurationManager, ConfigurationManager>();
+        }
+
```
Comment (tabas) [inline src/Reenbit.ChuckNorris.API/Extentions/DependencyInjectionConfig.cs:31]: -

File: src/Reenbit.ChuckNorris.DataAccess/ReenbitChuckNorrisDbContext.cs (hunk 1 above)
Comment (RomanSipyak) [reply src/Reenbit.ChuckNorris.DataAccess/ReenbitChuckNorrisDbContext.cs]: +

File: src/Reenbit.ChuckNorris.DataBase/JokeCategory.sql (hunk 2 above)
Comment (RomanSipyak) [reply src/Reenbit.ChuckNorris.DataBase/JokeCategory.sql]: +

File: src/Reenbit.ChuckNorris.DataBase/JokeCategory.sql (hunk 3 above)
Comment (RomanSipyak) [reply src/Reenbit.ChuckNorris.DataBase/JokeCategory.sql]: +

File: src/Reenbit.ChuckNorris.DataBase/Jokes.sql (hunk 5 above)
Comment (RomanSipyak) [reply src/Reenbit.ChuckNorris.DataBase/Jokes.sql]: +

File: src/Reenbit.ChuckNorris.Domain/Entities/JokeCategory.cs (hunk 6 above)
Comment (RomanSipyak) [reply src/Reenbit.ChuckNorris.Domain/Entities/JokeCategory.cs:13]: +

File: src/Reenbit.ChuckNorris.DataAccess/Mapping/JokeMap.cs (hunk 7 above)
Comment (RomanSipyak) [reply src/Reenbit.ChuckNorris.DataAccess/Mapping/JokeMap.cs]: +

**Pull request merged**